Added info about "props:card" option (#355)
mattermost/mattermost-developer-documentation: site/content/integrate/incoming-webhooks/_index.md

| attachments | [Message attachments](https://docs.mattermost.com/developer/message-attachments.html) used for richer formatting options. | If `text` is not set, yes |
| type | Sets the post `type`, mainly for use by plugins.<br> If not blank, must begin with "custom\_". | No |

> | props | Sets the post `props`, a JSON property bag for storing extra or meta data on the post.<br> Mainly used by other integrations accessing posts through the REST API.<br> The following keys are reserved: "from\_webhook", "override\_username", "override\_icon\_url", "override\_icon\_emoji", "webhook\_display\_name" and "attachments". | No |

Reviewer: I miss the `card` field as reserved word here.

Author: I added the 'card' field to reserved words.

> | props | Props `card` allows for extra information (markdown formatted text) to be sent to Mattermost that will only be displayed in the RHS panel after a user clicks on an 'info' icon displayed alongside the post.  The info icon cannot be customized and is only rendered visible to the user if there is `card` data passed into the message. This is only available in v5.14+. | No |

Reviewer 2: Should we incorporate it as an additional description to the `props` parameter on line 70?

Reviewer 2: Also, would we want to mention this is not supported on mobile?

Reviewer 3: Is this just available for webhooks or for posts in general?

Author: @hanzei Available for posts in General.  I'm updating the Zapier connector for example which is posting via the Rest API and has posted "card" data via the API (not webhook).  I guess we should also document it somewhere in the API docs too?

Reviewer 3: I'm not sure what you mean with "API docs". https://api.mattermost.com/?

I would like (3/5) to see a new page in https://developers.mattermost.com/integrate/ where all post parameters are described. What do you think about this?

Author: @hanzei - Yes, the API docs there.  You can now pass in "Props" with "card" to an HTTP endpoint.  How would you know that by looking at this docs page? 
![image](https://user-images.githubusercontent.com/915956/64437791-cb303600-d07b-11e9-9249-500c349a9460.png)

Curious - the page you'd like to see, would it be called "Global Post Parameters" or something like that?  Is there anywhere these Post-Params can be used *aside* from Incoming Webhooks and the REST API?   The reason I ask, is when looking through the dev.docs site, if I went to the "Integrate" area - you would likely either go to the Webhooks page (which we can ensure document all the params further) OR to the api.mattermost.com site which would then have the HTTP API Post documentation (and currently lacks any information about "card",  etc.).  Why would we expect someone to go to a separate page to find out about further options about the /api/v4/post endpoint?

Reviewer 2: On my original question, there are two 'props' parameters in the table, but should there be just one (both within the same `props` parameter)?

![image](https://user-images.githubusercontent.com/13119842/64462786-a2399080-d0cf-11e9-855a-13f823a79959.png)

Author: Will remove second box and merge

}
```

> An example request that would display additional info in the RHS panel, by passing markdown text into `props` as a `card` object would look like this:

Reviewer 2: ```suggestion
An example request displaying additional data in the right-hand side panel, by passing Markdown text into `props` as a `card` object would look like this:
```

> An example request displaying additional data in the right-hand side panel, by passing Markdown text into `props` as a `card` object would look like this:

Reviewer: I think is not clear enough here. I would say something like:

```suggestion
An example request displaying additional data in the right-hand side panel, by passing Markdown text into the `card` field of the `props` object would look like this:
```

```http
POST /hooks/xxx-generatedkey-xxx HTTP/1.1
Host: your-mattermost-server
Content-Type: application/json

{
  "channel": "town-square",
  "username": "Winning-bot",
  "text": "#### We won a new deal!"

>   "props": "{"card": "Salesforce Opportunity Information:\n\n [Opportunity Name](http://salesforce.com/OPPORTUNITY_ID)\n\n-Salesperson: **Bob McKnight** \n\n Amount: **$300,020.00**"}

Reviewer 2: ```suggestion
  "props": "{"card": "Salesforce Opportunity Information:\n\n [Opportunity Name](http://salesforce.com/OPPORTUNITY_ID)\n\n-Salesperson: **Bob McKnight** \n\n Amount: **$300,020.00**"}"
```

}
```

> When there is `props:card` data passed in via the webhook, the rendered message will display a small icon that expands the RHS panel to display the markdown that was passed in with the webhook:

Reviewer 2: ```suggestion
When there is `props:card` property attached to the webhook payload, the posted message displays a small info icon next to the timestamp. Clicking this icon expands the right-hand side panel to display the Markdown included in `props:card`:
```

> When there is `props:card` property attached to the webhook payload, the posted message displays a small info icon next to the timestamp. Clicking this icon expands the right-hand side panel to display the Markdown included in `props:card`:

Reviewer: ```suggestion
When there is `props` object with a `card` property attached to the webhook payload, the posted message displays a small info icon next to the timestamp. Clicking this icon expands the right-hand side panel to display the Markdown included in the `card` property:
```
